config/initializers/ostatus.rb

```diff
@@ -12,7 +12,7 @@
   config.x.use_s3       = ENV['S3_ENABLED'] == 'true'
 
   config.action_mailer.default_url_options = { host: web_host, protocol: https ? 'https://' : 'http://', trailing_slash: false }
-  config.x.streaming_api_base_url          = 'http://localhost:4000'
+  config.x.streaming_api_base_url          = "http://localhost:#{port.to_i + 100}"
```

Comment:
With foreman, $PORT is set at run time. Also, $PORT of the second and subsequent processes is the value obtained by adding 100 to the base port.
For example, when the base port is 3000, rails starts with port 3000 and node's server starts with port 3100.

```text:Procfile
web: bundle exec puma -C config/puma.rb             # $PORT is 3000
stream: yarn run start                              # $PORT is 3100
webpack: ./bin/webpack-dev-server --host 0.0.0.0    # $PORT is 3200
```
So streaming server can not be started with 4000.
Of course, it can be started with port 4000 if the port number is hard-coat.

```
stream: PORT=4000 yarn run start
```